services/claudie-operator/pkg/api/v1beta1/inputmanifest_helpers.go

```diff
@@ -38,14 +41,24 @@ func (im *InputManifest) GetNamespacedNameDashed() string {
 
 // GetSecretField takes an ENUM string type of SecretField, and returns the value
 // of the that field from the ProviderWithData struct
+// it is also validating if the string is a proper UTF-8 string
 func (pwd *ProviderWithData) GetSecretField(name SecretField) (string, error) {
 	if value, ok := pwd.Secret.Data[string(name)]; ok {
+		if !validateSecretUTF8(string(value)) {
+			return "", fmt.Errorf("field %s is not a valid UTF-8 string", name)
+		}
 		return string(value), nil
 	} else {
 		return "", fmt.Errorf("field %s not found", name)
 	}
 }
 
+// validateSecretUTF8 validates whether s consists entirely of valid UTF-8-encoded runes
+// Ref: https://github.com/berops/claudie/issues/1101#issuecomment-1820793262
+func validateSecretUTF8(data string) bool {
+	return utf8.ValidString(data)
+}
+
```

Comment:
This is not needed. Calling `utf8.ValidString()` can be done directly.

Reply:
> Did you also try it with invalid utf8 chars ?
Yes, I was testing it by running `cat /dev/urandom | head -n 3 | base64` and putting the value into the secret. It wasn't passing the validation.